dask/array/core.py

```diff
@@ -5549,29 +5557,32 @@ class BlockView:
     An instance of ``da.array.Blockview``
     """
 
-    def __init__(self, array: Array) -> BlockView:
+    def __init__(self, array: Array) -> None:
         self._array = array
 
     def __getitem__(
         self, index: int | Sequence[int] | slice | Sequence[slice]
-    ) -> Array:
+    ) -> Array:  # type: ignore
         from dask.array.slicing import normalize_index
 
         if not isinstance(index, tuple):
-            index = (index,)
-        if sum(isinstance(ind, (np.ndarray, list)) for ind in index) > 1:
+            new_index = (index,)
+        else:
+            new_index = index  # type: ignore
```

Comment:
mypy is telling that something's wrong here.
According to the definition:
``
index: int | Sequence[int] | slice | Sequence[slice]
``
after this block,
- if index was a int, new_index is now a ``tuple[int, ...]`` - good
- if index was a slice, new_index is now a ``tuple[slice, ...]`` - good
- if index was a Sequence but not a tuple, new_index is now a tuple of sequences. Which is actually intentionally in the implementation below.

If you defined new_index explicitly, you would save yourself a wealth of ``#type: ignore`` and benefit from more robust checks:
``new_index: tuple[int | slice, ...] | tuple[Sequence[int | slice]]``